### ISSUE TITLE
IPC command get_outputs returns "focused": false for all outputs

### ISSUE THREAD

**Issue by jplatte** [swaywm/sway#2685]
for a setup with two monitors:

```
$ swaymsg -t get_outputs | grep focused
     "focused": false,
     "focused": false,

$ swaymsg -t get_version
sway version 1.0-alpha.5-364-g15dadaaa (Sep 18 2018, branch 'makepkg')
```

**Comment by RyanDwyer** [comment]
This is intended. `focused` is only true for the single focused node (not the ancestors), so it can never be true on an output.

The i3 docs (https://i3wm.org/docs/ipc.html) confirm this:

> ... will result in eventually reaching the one node with `focused` set to true

And looking at the i3 code, it's only set for one node (`focused` is a global variable): https://github.com/i3/i3/blob/ac88d06632b6235e51d89806b7889ba5bfb9eb78/src/ipc.c#L420

**Comment by jplatte** [comment]
Okay. Then that field being there at all is a bit misleading though. I just checked and in i3, the field is not part of the `get_outputs` IPC command output.

**Comment by RyanDwyer** [comment]
Confirmed. An example of what they return is:

    {
        "name": "LVDS-1",
        "active": true,
        "primary": true,
        "rect": {
          "x": 0,
          "y": 0,
          "width": 1920,
          "height": 1080
        },
        "current_workspace": "3"
    },

I'd support removing irrelevant elements from our `get_outputs` reply.

**Comment by jplatte** [comment]
Also, as an aside: My main goal was to query the name of the currently focused output. I managed to do that after reading up on `jq`, which I had seen used for filtering out information from i3's ipc previously. This is what I came up for, for anyone looking to do the same thing:

`swaymsg -t get_tree | jq '.nodes[] | select(recurse(.nodes[]) | .focused).name'`

**Comment by her001** [comment]
May I ask what other properties besides "focused" are considered extraneous?

**Comment by ianyfan** [comment]
Probably anything that isn't in the i3 reply.

**Comment by progandy** [comment]
Not everything that isn't in i3 is extraneous. Some of the properties have to be included by sway to handle output configuration that is done by xrandr in Xorg.

**Comment by Moelf** [comment]
```
$ swaymsg -t get_outputs | grep focused
     "focused": true,
     "focused": false,
```

looks like now we have some useful outputs?

**Comment by ammgws** [comment]
@ianyfan It looks like this can be closed now.

**Comment by ianyfan** [comment]
I don't think so. Run `swaymsg -t get_outputs -r` in both sway and i3 and compare them.

**Comment by ammgws** [comment]
Ah OK I had conflated the issue of `focused` always returning false with the original issue of non-conformance with the i3 IPC.
